### PR TITLE
Detects shell and returns full path on windows

### PR DESCRIPTION
**Pull request by matthewdeanmartin**
tested with powershell, gitbash, cygwin, mingw.

The one pytest still passes, I didn't see any other checks to run.

Launching these shells by just invoking the full path is still going to be hit or miss, for example launching git bash requires

"C:\Program Files\Git\usr\bin\bash.exe" --login

and of course these are all backslash format & different shells have different strategies for handling full paths.

### CODE REVIEW

File: src/shellingham/nt.py [hunk 1]
```diff
@@ -106,6 +106,10 @@ def _get_executable(process_dict):
 def get_shell(pid=None, max_depth=6):
     """Get the shell that the supplied pid or os.getpid() is running in.
     """
+    import shellingham.windows as windows
+    # I don't know what the rest of the nt.py thing does.
+    return windows.get_shell(pid, max_depth)
```
Comment (uranusjr) [inline src/shellingham/nt.py:111]: The `nt` module contains code to detect the shell on NT platforms (IOW non-ancient Windows, technically called [Windows NT](https://en.wikipedia.org/wiki/Windows_NT), which Windows NT the operating system belongs to but is not exactly). So all the code in `windows.py` can simply be added there without a local import like this.

Comment (matthewdeanmartin) [reply src/shellingham/nt.py:111]: Do you want me to replace nt.py altogether? if get_shell is the only public function there, then that is the thing to do.

File: src/shellingham/nt.py [hunk 2]
```diff
@@ -1,128 +1,144 @@
...
+        size.value *= 2
+
+
+SHELLS = frozenset((
+    b'sh.exe', b'bash.exe', b'dash.exe', b'ash.exe',
+    b'csh.exe', b'tcsh.exe',
+    b'ksh.exe', b'zsh.exe', b'fish.exe',
+    b'cmd.exe', b'powershell.exe', b'pwsh.exe',
+    b'elvish.exe', b'xonsh.exe',
+))
```
Comment (uranusjr) [inline src/shellingham/nt.py:121]: Can this be calculated from `_core.SHELLS` instead? It’s probably not good to duplicate the list of shells in two places.

File: src/shellingham/nt.py [hunk 3]
```diff
@@ -1,128 +1,144 @@
-# Code based on the winappdbg project http://winappdbg.sourceforge.net/
-# (BSD License) - adapted from Celery by Dan Ryan (dan@danryan.co)
-# https://github.com/celery/celery/blob/2.5-archived/celery/concurrency/processes/_win.py
```
Comment (uranusjr) [inline src/shellingham/nt.py:3]: This needs to be kept somewhere to keep attribution.

**Comment by uranusjr** [comment]
🚢

**Comment by TBBle** [comment]
The README should probably be updated to remove the note about Windows only returning the shell executable name, without a path or extension.

**Comment by uranusjr** [comment]
PRs are always welcomed!

**Pull request merged**